snuba/utils/streams/consumer.py

```diff
@@ -97,6 +97,30 @@ def unsubscribe(self) -> None:
     def poll(self, timeout: Optional[float] = None) -> Optional[Message[TPayload]]:
         raise NotImplementedError
 
+    @abstractmethod
+    def pause(self, partitions: Sequence[Partition]) -> None:
+        """
+        Pause consuming from the provided partitions.
+
+        A partition that is paused will be automatically resumed during
+        reassignment. If partitions should remain paused across rebalances,
+        this should be implemented in the assignment callback.
+
+        If any provided partitions are not in the assignment set, an
+        exception will be raised and no parittions will be paused.
```

Comment:
typo: `no partitions`

Also I think it should be `If any of the provided partitions IS not in.....`

Reply:
Oops, thanks.

Regarding the grammar, wouldn't it be "are" since "partitions" is plural (compared with "If the partition is not in the assignment set…")